src/main/java/uk/gov/hmcts/juror/api/moj/service/JurorRecordServiceImpl.java

```diff
@@ -1263,6 +1269,30 @@ public void markResponded(String jurorNumber) {
         jurorHistoryRepository.save(history);
     }
 
+    @Override
+    public PaginatedList<FilterJurorRecord> searchForJurorRecords(JurorRecordFilterRequestQuery query) {
+
+        List<String> courts = SecurityUtil.getCourts();
+
+        return PaginationUtil.toPaginatedList(
+            jurorRepository.fetchFilteredJurorRecords(query, courts),
+            query,
+            query.getSortField(),
+            query.getSortMethod(),
+            tuple -> {
+                FilterJurorRecord.FilterJurorRecordBuilder builder = FilterJurorRecord.builder()
+                    .jurorNumber(tuple.get(QJuror.juror.jurorNumber))
+                    .jurorName(tuple.get(QJuror.juror.firstName.concat(" ").concat(QJuror.juror.lastName)))
+                    .postcode(tuple.get(QJuror.juror.postcode))
+                    .poolNumber(tuple.get(QJurorPool.jurorPool.pool.poolNumber))
+                    .courtName(tuple.get(QJurorPool.jurorPool.pool.courtLocation.name))
+                    .status(tuple.get(QJurorPool.jurorPool.status.statusDesc));
+                return builder.build();
+            },
+            500L
```

Comment:
I think we have a constant for this limit some where. Maybe worth checking

Reply:
maybe make this an app setting?

Reply:
From a quick search I can not see this constant anywhere so happy for this to be merged :)